sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/unity/identity/AzureDefaultTokenCredentialAutoConfiguration.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.autoconfigure.unity.identity;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.ChainedTokenCredentialBuilder;
+import com.azure.spring.autoconfigure.unity.AzureProperties;
+import com.azure.spring.identity.SpringAzureCliCredentialBuilder;
+import com.azure.spring.identity.SpringAzurePowerShellCredentialBuilder;
+import com.azure.spring.identity.SpringCredentialBuilderBase;
+import com.azure.spring.identity.SpringEnvironmentCredentialBuilder;
+import com.azure.spring.identity.SpringIntelliJCredentialBuilder;
+import com.azure.spring.identity.SpringManagedIdentityCredentialBuilder;
+import com.azure.spring.identity.SpringVisualStudioCodeCredentialBuilder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+
+import java.util.List;
+
+/**
+ * Auto-configuration for Azure Spring default token credential.
+ */
+@Configuration
+public class AzureDefaultTokenCredentialAutoConfiguration {
+
+    public static final int SPRING_ENV_CREDENTIAL_ORDER = 0;
+
+    @Bean
+    public AzureSpringConfiguration azureSpringConfiguration(AzureProperties azureProperties) {
+        return new AzureSpringConfiguration(azureProperties);
+    }
+
+    @ConditionalOnMissingBean
+    @Bean
+    @Order(SPRING_ENV_CREDENTIAL_ORDER)
+    public SpringEnvironmentCredentialBuilder springEnvironmentCredentialBuilder(AzureSpringConfiguration azureSpringConfiguration) {
+        return new SpringEnvironmentCredentialBuilder().credentialPropertiesProvider(azureSpringConfiguration);
+    }
+
+    @ConditionalOnMissingBean
+    @Bean
+    @Order(SPRING_ENV_CREDENTIAL_ORDER + 100)
```

Comment:
Maybe it's a good choice to implement `org.springframework.core.Ordered` to set the order.